lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,347 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+require('./browser.es6');
+require('./browser_api.es6');
+require('./web_interface.es6');
+
+foam.CLASS({
+  name: 'WebAPIs',
+  package: 'com.web.api',
+  requires: [
+    'com.web.api.Browser',
+    'com.web.api.BrowserAPI',
+    'com.web.api.WebInterface',
+    'foam.dao.EasyDAO',
+    'foam.mlang.sink.GroupBy',
+    'foam.dao.ArrayDAO',
+    'foam.mlang.ExpressionsSingleton',
+  ],
+  properties: [
+    {
+      name: 'browserAPIs',
+      documentation: `A DAO that contains pairs of browser Id
+        and interface Id.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserAPIDAO',
+          of: this.BrowserAPI,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'browsers',
+      documentation: `A DAO that contains browser names, versions
+        OS, OS versions and their browser Id.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserDAO',
+          of: this.Browser,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'interfaces',
+      documentation: `A DAO that contains interface names, API names,
+        and their interfaceId`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'InterfaceDAO',
+          of: this.WebInterface,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'mlang',
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'importAPI',
+      documentation: 'Import interface/API for a given version of browser.',
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The name of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'browserVersion',
+          documentation: `The version of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSName',
```

Comment:
Here and elsewhere: as with property names, don't start with capital unless it's a type. Prefer `osName`, `osVersion`, etc..